lib/dry/transaction/step_failure.rb

```diff
@@ -6,6 +6,16 @@ class StepFailure
       attr_reader :step
       attr_reader :value
 
+      # @api private
+      def self.call(step, value)
+        if self === value
+          value
+        else
+          yield
+          new(step, value)
```

Comment:
Is just to understand better what it is doing this `call` method. Why do we need a new `StepFailure` after publishing `:step_failed`?

Reply:
Because it worked like this:
![image](https://user-images.githubusercontent.com/802486/40849486-bd15f3d8-65ca-11e8-8ce2-6a43720aa188.png)

Reply:
Right. good point